crates/sui-adapter/src/execution_engine.rs

```diff
@@ -281,29 +281,44 @@ fn execute_transaction<
         };
         let cost_summary =
             temporary_store.charge_gas(gas_object_id, &mut gas_status, &mut result, gas);
+         // === begin SUI conservation checks ===
+    // For advance epoch transaction, we need to provide epoch rewards and rebates as extra
+    // information provided to check_sui_conserved, because we mint rewards, and burn
+    // the rebates. We also need to pass in the unmetered_storage_rebate because storage
+    // rebate is not reflected in the storage_rebate of gas summary. This is a bit confusing.
+    // We could probably clean up the code a bit.
         // Put all the storage rebate accumulated in the system transaction
         // to the 0x5 object so that it's not lost.
         temporary_store.conserve_unmetered_storage_rebate(gas_status.unmetered_storage_rebate());
-        #[cfg(debug_assertions)]
-        {
-            // Genesis transactions mint sui supply, and hence does not satisfy SUI conservation.
-            if !is_genesis_tx {
-                // For advance epoch transaction, we need to provide epoch rewards and rebates as extra
-                // information provided to check_sui_conserved, because we mint rewards, and burn
-                // the rebates. We also need to pass in the unmetered_storage_rebate because storage
-                // rebate is not reflected in the storage_rebate of gas summary. This is a bit confusing.
-                // We could probably clean up the code a bit.
-                if !Mode::allow_arbitrary_values() {
-                    // ensure that this transaction did not create or destroy SUI
-                    temporary_store
-                        .check_sui_conserved(advance_epoch_gas_summary)
-                        .unwrap();
-                }
-                // else, we're in dev-inspect mode, which lets you turn bytes into arbitrary
-                // objects (including coins). this can violate conservation, but it's expected
-                return (cost_summary, result);
-            }
-        }
+        if !is_genesis_tx {
+        // TODO: use node_config for this instead?
```

Comment:
@lxfind is this something that is easy to do (I couldn't figure out how to get at the node config in the authority code that calls this)?

Reply:
We would have to store the flag in authority and pas it down, I think